src/lib/apis/chatApi.ts

```diff
@@ -0,0 +1,361 @@
+  try {
+    if (!data.chatRoomId) return { chats: [], totalItemCount: 0 };
+
+    const params = new URLSearchParams();
+
+    Object.entries(data)
+      .filter(([_, v]) => v !== undefined)
+      .forEach(([k, v]) => {
+        if (Array.isArray(v)) {
+          v.forEach((value) => params.append(`${k}[]`, value));
+        } else {
+          params.append(k, String(v));
+        }
+      });
+
+    const response = await fetch(`/api/chat/get-chat?${params}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    if (!response.ok) {
+      const errorData = await response.json();
+      const error: FetchError = new Error(errorData.message || '');
+      error.status = response.status;
+      if (error.status === 404) {
+        return { chats: [], totalItemCount: 0 };
+      }
+      throw error;
+    }
+
+    const resData = await response.json();
+
+    return resData.data;
+  } catch (error) {
+    console.error('채팅 조회 에러', error);
+    throw error;
+  }
+};
+
+export const sendChat = async (
+  data: sendChatParams,
+  userType: userType,
+): Promise<Chat> => {
+  try {
+    if (!data.chatRoomId) {
+      throw 'chatRoomId undefined';
+    }
+
+    const response = await fetch(`/api/chat/send-chat?userType=${userType}`, {
+      method: 'POST',
+      credentials: 'include',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (!response.ok) {
+      const errorData = await response.json();
+      const error: FetchError = new Error(errorData.message || '');
+      error.status = response.status;
+      throw error;
+    }
+
+    const resData = await response.json();
+    return resData.data.chat;
+  } catch (error) {
+    console.error('채팅 전송 오류', error);
+    throw error;
+  }
+};
+
+export const getOpponentInfo = async (
+  userType: userType,
+  id: number,
+): Promise<OpponentInfo> => {
+  const url =
+    userType === 'lecturer'
+      ? `/api/post/instructor?id=${id}`
+      : `/api/users/get-info?userId=${id}`;
+
+  try {
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    if (!response.ok) {
+      const errorData = await response.json();
+      const error: FetchError = new Error(errorData.message || '');
+      error.status = response.status;
+      throw error;
+    }
+
+    const resData = await response.json();
+
+    return userType === 'lecturer'
+      ? {
+          id: resData.data.lecturerProfile.id,
+          nickname: resData.data.lecturerProfile.nickname,
+          profilImg:
+            resData.data.lecturerProfile.lecturerProfileImageUrl[0].url,
+        }
+      : {
+          id: resData.data.user.id,
+          nickname: resData.data.user.nickname,
+          profilImg: resData.data.user.userProfileImage?.imageUrl,
+        };
```

Comment:
```suggestion
    const { data } = await response.json();
    const { lecturerProfile, user } = data;
    
    const id = lecturerProfile.id || user.id;
    const nickname = lecturerProfile.nickname || user.nickname;
    const profileImg = lecturerProfileImageUrl[0].url || userProfileImage?.imageUrl;
    
    return { id, nickname, profileImg }
```
요 방법이 좀 더 가독성 좋을거 같습니당 
`profilImg` -> `profileImg` 오타 발견👀

Reply:
오케이 고마워 반영 할게 👍